common-npm-packages/docker-common-v2/registryauthenticationprovider/acrauthenticationtokenprovider.ts

```diff
@@ -45,4 +48,56 @@ export default class ACRAuthenticationTokenProvider extends AuthenticationTokenP
 
         return null;
     }
+
+    private _getACRToken(AADToken: string, retryCount: number, timeToWait: number): Q.Promise<string> {
+        let deferred = Q.defer<string>();
```

Comment:
Can this piece be written using async / await pattern which you have used elsewhere in this PR ?